node/core/approval-voting/src/criteria.rs

```diff
@@ -458,16 +472,18 @@ pub(crate) fn check_assignment_cert(
 	assignment: &AssignmentCert,
 	backing_group: GroupIndex,
 ) -> Result<DelayTranche, InvalidAssignment> {
+	use InvalidAssignmentReason as Reason;
+
 	let validator_public = config
 		.assignment_keys
 		.get(validator_index.0 as usize)
-		.ok_or(InvalidAssignment)?;
+		.ok_or(InvalidAssignment(Reason::ValidatorIndexOutOfBounds))?;
 
 	let public = schnorrkel::PublicKey::from_bytes(validator_public.as_slice())
-		.map_err(|_| InvalidAssignment)?;
+		.map_err(|_| InvalidAssignment(Reason::InvalidSignature))?;
```

Comment:
Wait a minute - that does not look like an `InvalidSignature`, but more like an internal error where we were not able to parse the public key for some reason.

Reply:
Good catch! I remember renaming it to InvalidKey, but somehow this change was lost in a rebase probably.